### ISSUE TITLE
Show 'has_new_inputs' in pipeline UI

### ISSUE THREAD

**Issue by vito** [concourse/concourse#3857]
## What challenge are you facing?

https://github.com/concourse/concourse/pull/3163 added an API field to the `Job` object: `has_new_inputs`. Example: the [`shipit` job](https://ci.concourse-ci.org/api/v1/teams/main/pipelines/concourse/jobs/shipit). We agreed to just merge the API change first and figure out the UI later.

## What would make this better?

Having an indicator in the pipeline UI so a user can know there are new inputs available for a job.

Suggestions:

* don't scare me into thinking my `shipit` job is running - make sure the UI indicator doesn't look like a 'running' or 'pending' halo from a distance
* don't be noisier than the existing pending/running animations

Notes:

* a job can be pending/running *and* marked `has_new_inputs` - we should probably show both, 
  * we may want to observe how this looks - e.g. when a build is auto-queued up for new inputs, it'll still be marked `has_new_inputs` until the next scheduling tick. this might end up looking weird. or it could end up looking *cool*.

Proposal: maybe have the lines leading to the job be highlighted? Rough example:

![image](https://user-images.githubusercontent.com/1880/57703831-cfe56000-762e-11e9-845c-c44fd8f4a366.png)

Maybe too bright (or maybe it just looks like that because there are so many lines), but yeah.

/cc @simonjohansson @Lindsayauchin

**Comment by marco-m** [comment]
@vito FYI https://discuss.concourse-ci.org/t/improved-ui-for-manually-triggered-jobs/146, which is the first discussion that led to #3163, shows a mock UI by @Lindsayauchin that in my opinion is clearer than the highlighted lines above :-)

**Comment by vito** [comment]
@marco-m Do you mean this?: https://discuss.concourse-ci.org/t/improved-ui-for-manually-triggered-jobs/146/5

**Comment by simonjohansson** [comment]
I like a combination of both @marco-m @vito , its quite neat to be able to see what resources are new from a previous run, especially in a bigger pipeline with lots of inputs like the one you show above.

**Comment by Lindsayauchin** [comment]
@vito @marco-m What if the change in style is on the resource box as a yellow outline?

![Screen Shot 2019-05-15 at 1 35 17 PM](https://user-images.githubusercontent.com/27746614/57796460-4f019380-7716-11e9-8fb6-b845f81ba6da.png)
![Screen Shot 2019-05-15 at 1 35 10 PM](https://user-images.githubusercontent.com/27746614/57796461-4f019380-7716-11e9-80c2-e1f3e6027f0f.png)

**Comment by marco-m** [comment]
@Lindsayauchin that would be fine for me.

**Comment by pn-santos** [comment]
I'm gonna suggest the outline be green, maybe blue-ish?

Yellow/orange has the "error/warning/in-progress" connotation imo.

**Comment by Lindsayauchin** [comment]
Bright Teal? :
![Screen Shot 2019-05-15 at 5 46 55 PM](https://user-images.githubusercontent.com/27746614/57811930-82094e80-7739-11e9-9833-8f435c5f6c94.png)
![Screen Shot 2019-05-15 at 5 47 11 PM](https://user-images.githubusercontent.com/27746614/57811933-846ba880-7739-11e9-823f-d70f5daf07fc.png)

**Comment by mockersf** [comment]
In the build view, the downward arrow for a resource is set to yellow when there is a new version. I think both indicators should have the same colour, but it can be changed in the build view.

Also, an input could be new for one of the jobs depending on it, but not for all of them, maybe because one is depending on several inputs or waiting on something else.
Let's say I have some fast tests that run on every commit to a repo, and heavy tests that only run during specific hours. The input won't stay new for the fast tests but will still be for the heavy tests.
For that reason, I think it would make more sense to change the line between the resource and the job rather than the resource itself

**Comment by Lindsayauchin** [comment]
@vito @mockersf maybe a combo of both? The lines leading to the job highlighted and the resource outlined? So that it is clear when the lines get hairy what resource has a new version?

<img width="1087" alt="Screen Shot 2019-05-15 at 10 31 36 PM" src="https://user-images.githubusercontent.com/27746614/57822200-471b1100-7761-11e9-8620-54ea05d750f0.png">

**Comment by vito** [comment]
Side note: right now the only data we have is a bool on the job. It would be nice to track which resource is the 'first occurrence' and only highlight its line or resource box. If we had that, I could definitely see @mockersf's point of using the same 'yellow' color, as it would naturally match the same meaning as the 'yellow down arrow' icon on the build page.

**Comment by Lindsayauchin** [comment]
@vito @mockersf  Makes sense to keep the yellow highlight consistent with the yellow arrow on the build page and doesn't look too jarring in the UI. Fits in nicely :) What do y'all think of outlining the resource box and the line?

![Screen Shot 2019-05-16 at 5 37 40 PM](https://user-images.githubusercontent.com/27746614/57889613-463abb80-7803-11e9-987f-ac18d07e41ca.png)
![Screen Shot 2019-05-16 at 5 37 54 PM](https://user-images.githubusercontent.com/27746614/57889618-4c309c80-7803-11e9-8f8a-3a3fc06fd23b.png)

**Comment by simonjohansson** [comment]
@Lindsayauchin 
> What do y'all think of outlining the resource box and highlighting the line with yellow?

I agree with @pn-santos 
> Yellow/orange has the "error/warning/in-progress" connotation imo.

**Comment by marco-m** [comment]
So what about the bright teal in  https://github.com/concourse/concourse/issues/3857#issuecomment-492836178 ?

**Comment by pn-santos** [comment]
Maybe the down arrow of the resources (= new version) could also be changed to bright teal so there is no inconsistency?

**Comment by marco-m** [comment]
> Maybe the down arrow of the resources (= new version) could also be changed to bright teal so there is no inconsistency?

ah, good point indeed!

**Comment by StevenArmstrong** [comment]
Yellow outline looks better for colour blind users like myself.

**Comment by simonjohansson** [comment]
@vito @Lindsayauchin any update on this? :)

**Comment by vito** [comment]
We have made another arbitrary decision.

Using colours is tricky 'cause we use that for statuses everywhere, and this is more of an annotation than a status, so we're gonna try adding a white border to resources in the pipeline view and change the yellow arrow to a white border on the step header. Similar to the red border we're adding for #4111 

Sound OK?

**Comment by Lindsayauchin** [comment]
![Screen Shot 2019-07-29 at 4 03 51 PM](https://user-images.githubusercontent.com/27746614/62078391-9201dc00-b21a-11e9-9696-edd33adb9fc8.png)
![Screen Shot 2019-07-29 at 4 04 06 PM](https://user-images.githubusercontent.com/27746614/62078394-94643600-b21a-11e9-8947-40290273b9f8.png)

**Comment by pivotal-jamie-klassen** [comment]
it feels right to me that we go all-in on the Tron aesthetic.

**Comment by simonjohansson** [comment]
> Using colours is tricky 'cause we use that for statuses everywhere, and this is more of an annotation than a status, so we're gonna try adding a white border to resources in the pipeline view

Im a bit confused by what is going on, lots of boxes... :)

For instance, `windows-rc` resource shows a white box on the `build-rc` job, even though the next jobs does not depend on it?

**Comment by vito** [comment]
@simonjohansson disregard that, i think we just went ham with the mockup :)

**Comment by Lindsayauchin** [comment]
sorry that was a mistake in a quick mock-up. We, the development team, are going to take a closer look at the clarity of this solution when we can see it rendered in the UI live. 

![Screen Shot 2019-08-08 at 12 15 07 PM](https://user-images.githubusercontent.com/27746614/62719542-367fdd00-b9d6-11e9-81af-6cc734a38055.png)

**Comment by simonjohansson** [comment]
Howdie, have anything happened regarding this?

**Comment by vito** [comment]
@simonjohansson Sorry, this has fallen by the wayside while we focus on archiving/favoriting pipelines and the v10 roadmap. I'll apply some labels that hopefully get it some attention from the community.

**Comment by taylorsilva** [comment]
# Issue Bankruptcy

This issue is being closed as part of 2025's **Issue Bankruptcy** declaration. [See here for details](https://blog.concourse-ci.org/posts/2025-01-06-concourse-2024-review/)

If you think this issue still has merit then feel free to open a new issue and reference this one.
